internal/ui/update.go

```diff
@@ -506,8 +642,50 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				cmd = m.updateTaskSequence()
 				cmds = append(cmds, cmd)
 			case contextBookmarksView:
-				url := m.contextBMList.SelectedItem().FilterValue()
+				url := m.taskBMList.SelectedItem().FilterValue()
 				cmds = append(cmds, openURL(url))
+			case prefixSearchView:
+				prefix := m.prefixSearchList.SelectedItem().FilterValue()
+				var taskList list.Model
+
+				switch m.activeTaskList {
+				case activeTasks:
+					taskList = m.taskList
+				case archivedTasks:
+					taskList = m.archivedTaskList
+				}
+
+				taskList.ResetFilter()
+				var tlCmd tea.Cmd
+
+				runes := []rune(prefix)
+
+				if len(runes) > 1 {
+					taskList.FilterInput.SetValue(string(runes[:len(runes)-1]))
+				}
+
+				taskList, tlCmd = taskList.Update(tea.KeyMsg{Type: -1, Runes: []int32{47}, Alt: false, Paste: false})
```

Comment:
TODO: Filtering this list programmatically by manually updating the list model definitely feels like a hack. Maybe this can be done better.